tests/test_constants.py

```diff
@@ -0,0 +1,39 @@
+from eth_utils import (
+        keccak,
+    )
+
+from rlp.codec import encode_raw
+
+from trie.constants import (
+        BLANK_NODE,
+        BLANK_HASH,
+        BLANK_NODE_HASH,
+    )
+
+from trie.smt import SparseMerkleTree as SMT
+
+
+def test_hash_constants():
+    assert BLANK_HASH == keccak(BLANK_NODE)
+    assert BLANK_NODE_HASH == keccak(encode_raw(b''))
+
+
+def test_smt256_empty_hashes():
+    DEPTH = 256  # Default depth is 32 bytes
+
+    # Start at the bottom
+    EMPTY_LEAF_NODE_HASH = BLANK_HASH
+    EMPTY_NODE_HASHES = [EMPTY_LEAF_NODE_HASH]
+
+    # More hashes the lower you go down the tree (to the root)
+    # NOTE: Did this with different code as a sanity check
+    for _ in range(DEPTH - 1):
+        EMPTY_NODE_HASHES.insert(0, keccak(EMPTY_NODE_HASHES[0] + EMPTY_NODE_HASHES[0]))
```

Comment:
FYI, `list.insert(0, thing)` is one of the slowest standard library operations around.  My recommendation would be to either:

- build the list in reverse order and then reverse it once you are done
- use the `toolz.accumulate` function which does exactly what you need here
- use a `collections.deque` which has fast poping and insertion at both ends.

Reply:
I will do this update, and try to squash some of it down.

Reply:
Used `collections.deque`, but didn't really make a difference